### ISSUE TITLE
torch.onnx support

### ISSUE THREAD

**Issue by leesunfreshing** [pyg-team/pytorch_geometric#728]
Hi, I understand pytorch_geometric is dedicated to pytorch. However could you kindly consider supporting torch.onnx in order to use the code on other platforms, such as paddle / mxnet?

**Comment by samvanstroud** [comment]
I had a look at converting PyG models using `torch.onnx.export` and the primary problem (at least for GATConv layers) seems to be the use of `torch_scatter` in the PyG [`softmax`](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/utils/softmax.py#L43) implementation. This requires `scatter_max` which is [not currently supported by ONNX](https://github.com/onnx/onnx/issues/4322).

@rusty1s, if we can get this added to ONNX we should be able to support conversion, right? I'm guessing the alternative of rewriting `softmax` without `torch_scatter` is not very palatable.

**Comment by rusty1s** [comment]
One can rewrite `softmax` via `scatter_max` but it may lead to numerical instabilities. One option you can try is to utilize the newly introduced `torch.scatter_reduce()` functionality in PyTorch 1.12 and replace it with the `torch_scatter` variant.

**Comment by samvanstroud** [comment]
Thanks, that's good to know. We still would need to support `torch.scatter_reduce()` in ONNX but I am hoping this will not be too difficult.

**Comment by fabianlim** [comment]
@samvanstroud so how will this work? lets say the onnx issue https://github.com/onnx/onnx/issues/4322 is resolved and ScatterElements min/max are supported. Then, we still require updates to `torch.onnx` to understand how to handle `torch_scatter` ?

**Comment by rusty1s** [comment]
I don't think so. The current idea is to deprecate `torch-scatter` soon and fully rely on `torch.scatter_reduce` for reductions. See https://github.com/pyg-team/pytorch_geometric/pull/5120.

**Comment by fabianlim** [comment]
@rusty1s Ic. so then a `max` reduction via `torch.scatter_reduce` operations can be supported by `ScatterElements` min/max as per https://github.com/onnx/onnx/issues/4322?

**Comment by rusty1s** [comment]
I am not a ONNX expert, but I think it should.

**Comment by Lewis-Lu** [comment]
Quite interested in this topic, any existing module need tackling ?
@samvanstroud 's proposition is inspiring.
